dart/lib/src/protocol/sentry_event.dart

```diff
@@ -226,6 +226,102 @@ class SentryEvent {
         debugMeta: debugMeta ?? this.debugMeta,
       );
 
+  /// Deserializes a [SentryEvent] from JSON [Map].
+  factory SentryEvent.fromJson(Map<String, dynamic> json) {
+    final breadcrumbsJson = json['breadcrumbs'] as List<dynamic>?;
+    final breadcrumbs = breadcrumbsJson?.map((e) {
+      return Breadcrumb.fromJson(e);
+    }).toList();
+
+    final stackTraceValuesJson = json['threads']?['values'];
+    Map<String, dynamic>? stackTraceValuesStacktraceJson;
+    if (stackTraceValuesJson?.isNotEmpty == true) {
+      stackTraceValuesStacktraceJson = {};
+      stackTraceValuesStacktraceJson =
+          stackTraceValuesJson?.first['stacktrace'];
+    }
+
+    final exceptionValuesJson = json['exception']?['values'];
+    Map<String, dynamic>? exceptionValuesItemJson;
+    if (exceptionValuesJson?.isNotEmpty == true) {
+      exceptionValuesItemJson = {};
+      exceptionValuesItemJson = exceptionValuesJson?.first;
+    }
+
+    final modulesJson = json['modules'] as Map<String, dynamic>?;
+    Map<String, String>? modules;
+    if (modulesJson != null) {
+      modules = {};
+      modulesJson.forEach((key, value) {
+        modules?[key] = value;
+      });
```

Comment:
I believe there should be a way to "addAll" instead of doing this, if it does, lets fix across all the usage of forEach

Reply:
This did not work in all cases, as some testw fail with the info that `Map<String, dynamic>` cannot be cast to `Map<String, String>`

Reply:
@ueman do you know. if theres a "clever" way to do that? looks like quite a good amount of boilerplate for "adding all"

Reply:
last case lets extract to a method that receives a Map as param and returns a Map as result, so we don't DRY for tags, modules, etc...

Reply:
> This did not work in all cases, as some test cases fail with the info that Map<String, dynamic> cannot be cast to Map<String, String>

Why though? Is it because it's nested deeper?

If that's not the case you should be aber to use https://api.dart.dev/stable/2.12.4/dart-core/Map/cast.html